src/org/jetbrains/plugins/scala/lang/refactoring/introduceVariable/ScopeSuggester.scala

```diff
@@ -40,10 +43,21 @@ object ScopeSuggester {
     }
 
     val isScriptFile = currentElement.getContainingFile.asInstanceOf[ScalaFile].isScriptFile()
+
+    val isContainTypeParameter = ScalaRefactoringUtil.isTypeContainsTypeParameter(currentElement)
```

Comment:
You need to collect all type parameters used in type element and check their owners.